projects/fusion-ui/components/table/common/services/table.service.ts

```diff
@@ -248,7 +249,54 @@ export class TableService {
     }
 
     isRowReadOnly(row: any): boolean {
-        return !!row.rowMetaData && !!row.rowMetaData.readonly;
+        return !!row.rowMetaData?.readonly;
+    }
+
+    isRowInRequest(row: any): boolean {
+        return row.rowMetaData?.inRequest;
+    }
+
+    toggleRowInRequest(row: any, isInRequest) {
+        this.insureRowMetaData(row);
+        row.rowMetaData.inRequest = isInRequest;
+    }
+
+    setRowspanColumnsData(rows: any[], columnsKeys: string[]) {
+        rows.forEach(row => {
```

Comment:
please do not manipulate rows source data

Reply:
Ok.